scm-ui/ui-components/src/markdown/MarkdownView.tsx

```diff
@@ -155,20 +164,148 @@ class MarkdownView extends React.Component<Props, State> {
+
+    if (rendererList.code) {
+      const codeRendererAdapter = function({ node, children }: any) {
+        children = children || [];
+        const renderProps = {
+          value: children[0],
+          language: Array.isArray(node.properties.className) ? node.properties.className[0].split("language-")[1] : ""
+        };
+        return React.createElement(rendererList.code, renderProps, ...children);
+      };
+      components.code = codeRendererAdapter;
+    }
+
+    if (rendererList.link) {
+      const linkRendererAdapter = function({ node, children }: any) {
+        const renderProps = {
+          href: node.properties.href || ""
+        };
+        children = children || [];
+        return React.createElement(rendererList.link, renderProps, ...children);
+      };
+      components.a = linkRendererAdapter;
+    }
+
+    if (rendererList.heading) {
+      const createHeadingRendererAdapter = (level: number) => ({ node, children }: any) => {
+        const renderProps = {
+          id: node.properties.id,
+          level,
+          permalink
+        };
+        children = children || [];
+        return React.createElement(rendererList.heading, renderProps, ...children);
+      };
+      components.h1 = createHeadingRendererAdapter(1);
+      components.h2 = createHeadingRendererAdapter(2);
+      components.h3 = createHeadingRendererAdapter(3);
+      components.h4 = createHeadingRendererAdapter(4);
+      components.h5 = createHeadingRendererAdapter(5);
+      components.h6 = createHeadingRendererAdapter(6);
+    }
+
+    if (rendererList.break) {
+      components.br = rendererList.break;
+    }
+
+    if (rendererList.delete) {
+      components.del = rendererList.delete;
+    }
+
+    if (rendererList.emphasis) {
+      components.em = rendererList.emphasis;
+    }
+
+    if (rendererList.blockquote) {
+      components.blockquote = rendererList.blockquote;
+    }
+
+    if (rendererList.image) {
+      components.img = rendererList.image;
+    }
+
+    if (rendererList.list) {
+      components.ol = rendererList.list;
+      components.ul = rendererList.list;
+    }
+
+    if (rendererList.listItem) {
+      components.li = rendererList.listItem;
+    }
+
+    if (rendererList.paragraph) {
+      components.p = rendererList.paragraph;
+    }
+
+    if (rendererList.strong) {
+      components.strong = rendererList.strong;
+    }
+
+    if (rendererList.table) {
+      components.table = rendererList.table;
+    }
+
+    if (rendererList.tableHead) {
+      components.thead = rendererList.tableHead;
+    }
+
+    if (rendererList.tableBody) {
+      components.tbody = rendererList.tableBody;
+    }
+
+    if (rendererList.tableRow) {
+      components.tr = rendererList.tableRow;
+    }
+
+    if (rendererList.tableCell) {
+      components.td = rendererList.tableCell;
+      components.th = rendererList.tableCell;
+    }
+
+    if (rendererList.thematicBreak) {
+      components.hr = rendererList.thematicBreak;
+    }
```

Comment:
Also extract this whole block as a function to a new file? I would like to reduce this components size